inc/notificationtargetorder.class.php

```diff
@@ -490,6 +495,33 @@ public function addSpecificTargets($data, $options) {
          case self::SUPERVISOR_DELIVERY_GROUP:
             $this->addForGroup(1, $this->obj->fields['groups_id_delivery']);
             break;
+        case self::SUPPLIER:
+            $this->getAddressByType("suppliers", $this->obj->fields['id']);
+            break;
+        case self::CONTACT:
+            $this->getAddressByType("contacts", $this->obj->fields['id']);
+            break;
+      }
+   }
+
+      /**
+    * Get supplier or contact related to the order
+    *
+    * @param $recipient_type  recipient type(suppliers or contacts)
+    *
+    * @param $order_id        order id
+    *
+   **/
+   function getAddressByType($recipient_type, $order_id) {
+      global $DB;
+      $query = "SELECT glpi_".$recipient_type.".email AS email,glpi_".$recipient_type.".name AS name
```

Comment:
It would be better to use querying syntax as described [here](https://glpi-developer-documentation.readthedocs.io/en/master/devapi/database/dbiterator.html) instead of using raw SQL.